simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/states/AvailableState.java

```diff
@@ -43,28 +45,31 @@ public void onPlug(int evseId, int connectorId) {
 
         StationMessageSender stationMessageSender = stateManager.getStationMessageSender();
 
+        CompletableFuture<UserMessageResult> future = new CompletableFuture<>();
         evse.plug(connectorId);
         stationMessageSender.sendStatusNotificationAndSubscribe(evse, evse.findConnector(connectorId), (statusNotificationRequest, statusNotificationResponse) -> {
-
             OptionList<TxStartStopPointVariableValues> startPoints = stateManager.getStationStore().getTxStartPointValues();
             if (startPoints.contains(TxStartStopPointVariableValues.EV_CONNECTED) && !startPoints.contains(TxStartStopPointVariableValues.POWER_PATH_CLOSED)) {
                 String transactionId = TransactionIdGenerator.getInstance().getAndIncrement();
                 evse.createTransaction(transactionId);
 
                 stationMessageSender.sendTransactionEventStart(evseId, connectorId, TransactionEventRequest.TriggerReason.CABLE_PLUGGED_IN, TransactionData.ChargingState.EV_DETECTED);
             }
+            future.complete(UserMessageResult.SUCCESSFUL);
         });
 
         stateManager.setStateForEvse(evseId, new WaitingForAuthorizationState());
+        return future;
     }
 
     @Override
-    public void onAuthorize(int evseId, String tokenId) {
+    public CompletableFuture<UserMessageResult> onAuthorize(int evseId, String tokenId) {
         StationMessageSender stationMessageSender = stateManager.getStationMessageSender();
         StationStore stationStore = stateManager.getStationStore();
 
         log.info("in authorizeToken {}", tokenId);
 
+        CompletableFuture<UserMessageResult> future = new CompletableFuture<>();
         stationMessageSender.sendAuthorizeAndSubscribe(tokenId, singletonList(evseId), (request, response) -> {
             if (response.getIdTokenInfo().getStatus() == IdTokenInfo.Status.ACCEPTED) {
```

Comment:
IMHO time to fetch lambda body to a method ...
maybe some more simplification will arrive